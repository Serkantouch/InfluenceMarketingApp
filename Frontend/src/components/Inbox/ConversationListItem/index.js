import React, {useEffect, Component} from 'react';
import Badge from '@material-ui/core/Badge';

import './ConversationListItem.css';

class ConversationListItem extends Component {

  conversationClick = () => {
    this.props.onClick();
  }

  render() {
    let { photo, name, text, unreadCount , role} = this.props.data;
    let roleBadge = (role !== null && role !== undefined) ? <span title={role} class="badge badge-danger ml-2">{role[0]}</span> : null;
    if (photo === null || photo === undefined) photo = window.location.origin + '/dummy.png';
    let className = "conversation-list-item"
    if (this.props.isActive) {
      className+="-active"
    }
    return (
      <div className={className} onClick={this.conversationClick}>
        <img className="conversation-photo" src={photo} alt="conversation" />
        <div className="conversation-info">
          <h1 className="conversation-title">{name}</h1>
          {/*<p className="conversation-snippet">{text}</p>*/}
        </div>
        <Badge style={{ marginLeft: "auto", marginRight: "10px" }} badgeContent={unreadCount} color="primary"></Badge>
        
        {roleBadge}
      </div>
    );
  }
}

export default ConversationListItem;