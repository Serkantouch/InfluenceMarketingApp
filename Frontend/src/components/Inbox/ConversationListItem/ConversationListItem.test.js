// Link.react.test.js
import React from 'react';
//import cookie from 'react-cookies'

import ConversationListItem from '.';
import renderer from 'react-test-renderer'


it('should render correctly', () => {
  //cookie.save("cookie_user", '013741480', { path: "/" });  
    const component = renderer.create(<ConversationListItem data={{
        name: "John Doe",
        unreadCount: 10,
        photo: null
    }}/>);
    expect(component).toMatchSnapshot();
});



