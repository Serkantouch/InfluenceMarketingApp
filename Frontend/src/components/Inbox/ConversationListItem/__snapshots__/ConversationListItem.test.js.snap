// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="conversation-list-item"
  onClick={[Function]}
>
  <img
    alt="conversation"
    className="conversation-photo"
    src="http://localhost/dummy.png"
  />
  <div
    className="conversation-info"
  >
    <h1
      className="conversation-title"
    >
      John Doe
    </h1>
  </div>
  <span
    className="MuiBadge-root"
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": "10px",
      }
    }
  >
    <span
      className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary"
    >
      10
    </span>
  </span>
</div>
`;
